HttpFs/HttpFs.fs

```diff
@@ -863,17 +909,135 @@ module Client =
+      match wex.Status with
+      | WebExceptionStatus.Success ->
+        failwith "The Success status should not be used when exceptions are thrown"
+
+      | WebExceptionStatus.NameResolutionFailure ->
+        TcpException NameResolutionFailure
+
+      | WebExceptionStatus.ConnectFailure ->
+        TcpException ConnectFailure
+
+      | WebExceptionStatus.ConnectionClosed
+      | WebExceptionStatus.ReceiveFailure
+      | WebExceptionStatus.SendFailure ->
+        TcpException SocketClosed
+
+      | WebExceptionStatus.PipelineFailure ->
+        failwith "HttpFs has not been tested with HTTP pipelining; got WebExceptionStatus.PipelineFailure"
+
+      | WebExceptionStatus.RequestCanceled ->
+        Cancelled
+
+      | WebExceptionStatus.ProtocolError ->
+        failwith "HttpFs handles HTTP status codes, not as exceptions, but as normal responses"
+
+      | WebExceptionStatus.TrustFailure ->
+        TlsException BrokenTrust
+
+      | WebExceptionStatus.SecureChannelFailure ->
+        TlsException SecureChannelFailure
+
+      | WebExceptionStatus.ServerProtocolViolation ->
+        HttpException InvalidServerResponse
+
+      | WebExceptionStatus.KeepAliveFailure ->
+        HttpException KeepAliveClosed
+
+      | WebExceptionStatus.Pending ->
+        failwith "HttpFs should not call operations out of order on the underlying HTTP libraries"
+
+      | WebExceptionStatus.Timeout ->
+        failwith "HttpFs handles timeout by not committing to the returned Alt and using Hopac.Global.timeOut"
+
+      | WebExceptionStatus.ProxyNameResolutionFailure ->
+        TcpException NameResolutionFailureProxy
+
+      | WebExceptionStatus.UnknownError ->
+        match wex.InnerException with
+        | null ->
+          failwith "CLR returned 'Unknown Error' but there's no inner exception, here's the inner message '%s'"
+                   wex.Message
+
+        | :? System.Net.Sockets.SocketException as se ->
+          TcpException (SocketException se)
+
+        | :? System.IO.IOException as ioe ->
+          TcpException (IOException ioe)
+
+        | e ->
+          raise <| Exception(sprintf "Unknown exception from the CLR - %s" wex.Message, e)
+
+      | WebExceptionStatus.MessageLengthLimitExceeded ->
+        HttpException TooLongResponseHeaders
+
+      | WebExceptionStatus.RequestProhibitedByCachePolicy ->
+        failwith "HttpFs doesn't support cache policies"
+
+      | WebExceptionStatus.RequestProhibitedByProxy ->
+        TcpException NameResolutionFailureProxy
+
+      | WebExceptionStatus.CacheEntryNotFound ->
+        failwith "using CacheEntry with HttpFs not supported"
+
+      | other ->
+        failwith "There are only 21 options in this enum, something the F# compiler should be aware of"
+
+    let tryCommunicate (request : WebRequest) (xJ : Job<_>) fWebExnResp : Alt<Choice<'a, Error>> =
+      Alt.withNackJob <| fun nack ->
+        let rI = IVar ()
+
+        let rJ =
```

Comment:
Is `rJ` an unused variable? I assume you want to start a new job here to write the result to `rI`.